tools/Config.mk

```diff
@@ -42,13 +42,13 @@ else ifeq ($(V),)
   BASHCMD := $(shell command -v bash 2> /dev/null)
   ifneq ($(BASHCMD),)
     export SHELL=$(BASHCMD)
-    export ECHO_BEGIN=@echo -ne "\033[1K\r"
+    export ECHO_BEGIN=@echo " " -ne "\033[1K\r"
     export ECHO_END=$(ECHO_BEGIN)
   endif
 endif
 
 ifeq ($(ECHO_BEGIN),)
-  export ECHO_BEGIN=@echo
+  export ECHO_BEGIN=@echo " "
```

Comment:
can we add space in caller site? like this:
$(ECHO_BEGIN) "CPP: $1->$2"

Reply:
I think that is better

Reply:
>can we add space in caller site? like this:

@xiaoxiang781216 
In this case, we need to modify 3 files which include apps/Application.mk

```
find . -type f -exec grep --color -nH -e ECHO_BEGIN {} +
./nuttx/tools/Config.mk:45:    export ECHO_BEGIN=@echo " " -ne "\033[1K\r"
./nuttx/tools/Config.mk:46:    export ECHO_END=$(ECHO_BEGIN)
./nuttx/tools/Config.mk:50:ifeq ($(ECHO_BEGIN),)
./nuttx/tools/Config.mk:51:  export ECHO_BEGIN=@echo " "
./nuttx/tools/Config.mk:283:    $(ECHO_BEGIN)"CPP: $1->$2 "
./nuttx/tools/Config.mk:301:    $(ECHO_BEGIN)"CC: $1 "
./nuttx/tools/Config.mk:320:    $(ECHO_BEGIN)"CXX: $1 "
./nuttx/tools/Config.mk:339:    $(ECHO_BEGIN)"RUSTC: $1 "
./nuttx/tools/Config.mk:358:    $(ECHO_BEGIN)"ZIG: $1 "
./nuttx/tools/Config.mk:384:    $(ECHO_BEGIN)"AS: $1 "
./nuttx/tools/Config.mk:393:    $(ECHO_BEGIN)"IN: $1 -> $2 "
./nuttx/tools/Config.mk:415:    $(ECHO_BEGIN)"AR (add): ${shell basename $(1)} $(2) "
./nuttx/arch/sim/src/Makefile:288:      $(Q) $(ECHO_BEGIN)"CC:  $<"
./apps/Application.mk:112:      $(ECHO_BEGIN)"AS: $1 "
./apps/Application.mk:118:      $(ECHO_BEGIN)"CC: $1 "
./apps/Application.mk:124:      $(ECHO_BEGIN)"CXX: $1 "
./apps/Application.mk:130:      $(ECHO_BEGIN)"RUSTC: $1 "
./apps/Application.mk:136:      $(ECHO_BEGIN)"ZIG: $1 "
./apps/Application.mk:143:      $(ECHO_BEGIN)"LD: $2 "
```

Reply:
Then maybe adding back one space in `export ECHO_BEGIN=@echo ` can be a simplest solution

Reply:
Yes, it's better even the change will touch more files. Since:

1. It avoid the people/editor delete the tail space again
2. The reference point more like a normal shell command

It's very ugly that there is no space between $(ECHO_BEGIN) and "CPP: $1->$2 "

Reply:
let's restore compilation stability and add a space in all the places as a separate PR

Reply:
I can't repo this issue locally, ci server looks fine. how can I repo @masayuki2009?

Reply:
The reason for adding spaces here is mainly to let the echo string be displayed at the beginning of the line, e.g:

```bash
export ECHO_BEGIN=@echo -ne "\033[1K\r"
export ECHO_BEGIN=@echo

$(ECHO_BEGIN)"CC: $1 "
```

after expand:
```bash
echo -ne "\033[1K\r")"CC: $1 "
echo "CC: $1 "
```

```bash
CC:  mqueue/mq_timedsend.c
```

But if add the space after ECHO_BEGIN:

```bash
export ECHO_BEGIN=@echo -ne "\033[1K\r"
export ECHO_BEGIN=@echo

$(ECHO_BEGIN) "CC: $1 "
```

after expand:
```bash
echo -ne "\033[1K\r") "CC: $1 "    <--- After escape, there will be an extra space at the beginning of the line
echo "CC: $1 "
```

```bash
 CC:  mqueue/mq_timedsend.c   <--  echo -ne "\033[1K\r") "CC: $1 "  
CC:  mqueue/mq_timedsend.c   <-- echo "CC: $1 " 
```

Reply:
Ok, understand.